sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -718,28 +739,25 @@ private BasicDocumentRevision updateDocument(SQLDatabase db,String docId, String
         }
         CouchUtils.validateRevisionId(prevRevId);
 
-        DocumentUpdated documentUpdated = null;
+        try {
             BasicDocumentRevision preRevision = this.getDocumentInQueue(db, docId, prevRevId);
-            if (preRevision == null) {
-                throw new IllegalArgumentException("The document trying to update does not exist.");
-            }
 
             if (!preRevision.isCurrent()) {
                 throw new ConflictException("Revision to be updated is not current revision.");
             }
 
-            this.setCurrent(db,preRevision, false);
-            String newRevisionId = this.insertNewWinnerRevision(db,body, preRevision, copyAttachments);
-            BasicDocumentRevision newRevision = this.getDocumentInQueue(db, preRevision.getId(),
-                    newRevisionId);
+            this.setCurrent(db, preRevision, false);
+            String newRevisionId = this.insertNewWinnerRevision(db, body, preRevision, copyAttachments);
+            return this.getDocumentInQueue(db, preRevision.getId(), newRevisionId);
 
-            documentUpdated = new DocumentUpdated(preRevision, newRevision);
-            return newRevision;
+        } catch (DocumentNotFoundException e){
+            throw new IllegalArgumentException("The document trying to update does not exist.");
```

Comment:
Also, is IllegalArgumentException correct? There's nothing wrong with the argument of itself, DocNotFound seems more logical here.